cartridge/failover.lua

```diff
@@ -391,6 +396,62 @@ local function reconfigure_all(active_leaders)
     confapplier.set_state('RolesConfigured')
 end
 
+local function instance_secured()
+    if vars.client:check_quorum() then
+        return true
+    end
+
+    local confapplier = require('cartridge.confapplier')
+    local instance_uuid = confapplier.get_instance_uuid()
+    local replicaset_uuid = confapplier.get_replicaset_uuid()
+
+    local topology_cfg = confapplier.get_readonly('topology')
+    local replicaset_members = topology_cfg.replicasets[replicaset_uuid].master -- TODO consider old format
+
+    local refined_uri = topology.refine_servers_uri(topology_cfg)
+    local members = membership.members()
+
+    if #replicaset_members == 1 then
+        return false
+    end
+
+    local any_replica_available = false -- TODO all_rw, no replica
+    for _, replica_uuid in ipairs(replicaset_members) do
+        if replica_uuid ~= instance_uuid
+        and members[refined_uri[replica_uuid]].status == 'alive' then
+             any_replica_available = true
+        end
+    end
+
+    return any_replica_available
+end
+
+monitor_state_provider = function()
+    local fencing_timeout = 0.2 -- TODO
+    local fencing_step = 0.1 -- TODO
+    local fencing_deadline = fiber.time() + fencing_timeout
+
+    while fiber.time() < fencing_deadline do
+        if (instance_secured()) then
+            fencing_deadline = fiber.time() + fencing_timeout
+        end
+
+        fiber.sleep(fencing_step)
+    end
+
+    vars.cache.is_vclockkeeper = false;
+    vars.cache.is_rw = false
+    box.cfg{read_only = not vars.cache.is_rw}
+
+    do
+        local task = fiber.new(reconfigure_all, vars.cache.active_leaders)
+        local id = task:id()
+        task:name('cartridge.fencing.task')
+        vars.schedule[id] = task
+        log.info('Fencing triggered, reapply scheduled (fiber %d)', id)
+    end -- TODO new tasks
```

Comment:
The first thing to be done after fencing is triggered is to apply config with `is_master = false`. But without unsetting active_leaders the task will hang forever.

Also all pending tasks should be cancelled beforehead.

Reply:
and done